dev/com.ibm.ws.ejbcontainer.bindings_fat/test-applications/NoInterfaceBndWeb.war/src/com/ibm/ws/ejbcontainer/bindings/noInterface/bnd/web/NoInterfaceBindingAbstractServlet.java

```diff
@@ -0,0 +1,486 @@
+
+    /**
+     * Tests the default bindings for beans with the following configurations:
+     *
+     * <ul>
+     * <li> A bean that is only exposed through the No-Interface view.
+     * <li> A bean that has both a No-Interface view and Local business interface.
+     * <li> A bean that has both a No-Interface view and Remote business interface.
+     * <li> A bean that has both a No-Interface view and Local component interface.
+     * </ul>
+     *
+     * For configurations with multiple interfaces, all interfaces are tested. <p>
+     *
+     * Note : since multiple beans are using the same implementation class,
+     * the short default binding should result in an
+     * AmbiguousEJBReferenceException. All other long default bindings
+     * should work fine.
+     **/
+    @Test
+    public void testNoInterfaceDefaultBindings() throws Exception {
+        String beanNameLocal = beanName + "Local";
+        String beanNameRemote = beanName + "Remote";
+        String beanNameComp = beanName + "Component";
+        ComplexNoInterfaceBean bean = null;
+        ivContext = (Context) new InitialContext().lookup("");
+
+        // -----------------------------------------------------------------------
+        // Lookup short default No-Interface bean - Ambiguous
+        // -----------------------------------------------------------------------
+        svLogger.info("Lookup short default No-Interface bean - Ambiguous");
+        try {
+            bean = (ComplexNoInterfaceBean) ivContext.lookup("ejblocal:" + beanInterface);
+            fail("Ambiguous No-Interface short default found");
+        } catch (NamingException nex) {
+            // Throwable cause = nex.getCause();
+            // if (cause instanceof AmbiguousEJBReferenceException) {
+            // svLogger.info("lookup of short default failed as expected : " +
+            // cause.getClass().getName() + " : " +
+            // cause.getMessage());
+            // } else {
+            svLogger.info(nex.getClass().getName() + " : " + nex.getMessage());
+            nex.printStackTrace();
+            fail("Ambiguous No-Interface short default lookup failed in an " +
+                 "unexpected way : " + nex.getClass().getName() + " : " +
+                 nex.getMessage());
+            // }
+        }
+
+        // -----------------------------------------------------------------------
+        // Lookup long default of No-Interface bean
+        // -----------------------------------------------------------------------
+        svLogger.info("Lookup long default of No-Interface bean");
+        bean = (ComplexNoInterfaceBean) ivContext.lookup("ejblocal:" + APP_NAME + "/" + MOD_NAME +
+                                                         "/" + beanName + "#" + beanInterface);
+
+        // Verify the reference is valid by using it....
+        assertEquals(beanName, bean.getBeanName());
```

Comment:
This is kind of a big ask to add during a test port, but I hate asserts with no messages because if it fails you are just going to get 'LocalBean != LocalBean2' or 'LocalBean != null' with no explanation unless you dig into the tests. Something as simple as assertEquals("Lookup long default of No-Interface bean failed", beanName, bean.getBeanName()); would be pretty helpful

Reply:
Thanks for catching; that bothers me, too.  I've made some updates.  Let me know what you think.